rust/agama-lib/src/http_client.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::Context;
+use reqwest::{header, Client, Response};
+use serde::de::DeserializeOwned;
+
+use crate::{auth::AuthToken, error::ServiceError};
+
+pub struct HTTPClient {
+    client: Client,
+    pub base_url: String,
+}
+
+const API_URL: &str = "http://localhost/api";
+
+impl HTTPClient {
+    // if there is need for client without authorization, create new constructor for it
+    pub async fn new() -> Result<Self, ServiceError> {
+        let token = AuthToken::find().context("You are not logged in")?;
+
+        let mut headers = header::HeaderMap::new();
+        let value = header::HeaderValue::from_str(format!("Bearer {}", token).as_str())
+            .map_err(|e| ServiceError::NetworkClientError(e.to_string()))?;
+
+        headers.insert(header::AUTHORIZATION, value);
+
+        let client = Client::builder().default_headers(headers).build()?;
+
+        Ok(Self {
+            client,
+            base_url: API_URL.to_string(), // TODO: add support for remote server
+        })
+    }
+
+    const NO_TEXT: &'static str = "No text";
+    // Simple wrapper around Response to get target type.
+    // For more advanced usage use directly get method
+    pub async fn get_type<T: DeserializeOwned>(&self, path: &str) -> Result<T, ServiceError> {
```

Comment:
I like the idea but `get_type` is kind of confusing (am I getting the "type"?).

In any case, I would call this method `get` and rename the `get` method below to something like `get_raw` as I expect us to use this more often.

Reply:
agreed. I am even thinking about naming this one as `get_object` but it is right, that plain get here and maybe `get_response` would be better